pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/ByteBufferSchema.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.client.impl.schema;
+
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.common.schema.SchemaInfo;
+
+import java.nio.ByteBuffer;
+
+public class ByteBufferSchema implements Schema<ByteBuffer> {
+    @Override
+    public byte[] encode(ByteBuffer message) {
+        return message.array();
```

Comment:
This would only work on heap-based ByteBuffer, though the main use case for ByteBuffer is to mask whether the bytes are in JVM heap or direct memory.  

I think we should instead change the `Schema` interface to use `ByteBuffer` internally instead of `byte[]` so that we can easily support both.